src/screens/AddCash/AddCash.js

```diff
@@ -0,0 +1,193 @@
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import React from 'react';
+import { ScrollView, Keyboard, Platform } from 'react-native';
+import t from 'translations/translate';
+import styled from 'styled-components/native';
+import { useNavigation } from 'react-navigation-hooks';
+
+// utils
+import { fontSizes, appFont } from 'utils/variables';
+import { isValidFiatValue } from 'utils/validators';
+import { getCurrencySymbol, hasTooMuchDecimals } from 'utils/common';
+import { openInAppBrowser } from 'utils/inAppBrowser';
+import { rampWidgetUrl } from 'utils/fiatToCrypto';
+import { getActiveAccount, getAccountAddress, isSmartWalletAccount } from 'utils/accounts';
+import { useThemeColors } from 'utils/themes';
+
+// compomnents
+import { Container } from 'components/modern/Layout';
+import Button from 'components/modern/Button';
+import TextInput from 'components/TextInput';
+import HeaderBlock from 'components/HeaderBlock';
+import Text from 'components/modern/Text';
+import Toast from 'components/Toast';
+import Modal from 'components/Modal';
+import BuyCryptoAccountNotActiveModal from 'components/BuyCryptoAccountNotActiveModal';
+import ValueInputAccessoryHolder, {
+  INPUT_ACCESSORY_NATIVE_ID,
+} from 'components/ValueInputAccessory/ValueInputAccessoryHolder';
+
+// selectors
+import { useFiatCurrency, accountsSelector, useRootSelector } from 'selectors';
+import { useUser } from 'selectors/user';
+
+const AddCash = () => {
+  const navigation = useNavigation();
+  const [value, setValue] = React.useState('0');
+  const fiatCurrency = useFiatCurrency();
+  const colors = useThemeColors();
+  const currencySymbol = getCurrencySymbol(fiatCurrency);
+  const accounts = useRootSelector(accountsSelector);
+  const user = useUser();
+
+  const getCryptoPurchaseAddress = (): string | null => {
+    const activeAccount = getActiveAccount(accounts);
+
+    if (!activeAccount || !isSmartWalletAccount(activeAccount)) {
+      Modal.open(() => <BuyCryptoAccountNotActiveModal />);
+      return null;
+    }
+
+    return getAccountAddress(activeAccount);
+  };
+
+  const handleChangeText = (text: string) => {
+    const amount = text.replace(currencySymbol, '');
+    if (hasTooMuchDecimals(amount, 2)) return;
+    const updatedAmount = amount.replace(/^0+/, '');
+    setValue(updatedAmount);
+  };
+
+  const openUrl = async (url: string | null) => {
+    if (url) {
+      await openInAppBrowser(url).catch(showServiceLaunchError);
+    } else {
+      showServiceLaunchError();
+    }
+  };
+
+  const showServiceLaunchError = () => {
+    Toast.show({
+      message: t('toast.cryptoPurchaseLaunchFailed'),
+      emoji: 'hushed',
+      supportLink: true,
+    });
+  };
+
+  const handleUseValue = async (accessoryValue: string) => {
+    Keyboard.dismiss();
+    setValue(accessoryValue);
+  };
+
+  const onInputBlur = () => {
```

Comment:
These two onInputBlur/Focus are some simple that they could be inlinded in inputProps definition below

Reply:
Done.